backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/VmHandler.java

```diff
@@ -1615,21 +1615,27 @@ private VmNameValidator getVmNameValidator(OriginType originType) {
         }
     }
 
-    public ValidationResult validateCpuPinningPolicy(VmBase vmBase, boolean numaSet, Version version) {
+    public ValidationResult validateCpuPinningPolicy(VmBase vmFromParams, Version version) {
+        return validateCpuPinningPolicy(vmFromParams, null, version);
+    }
+
+    public ValidationResult validateCpuPinningPolicy(VmBase vmFromParams, VmBase vmFromDb, Version version) {
         ValidationResult result = ValidationResult.VALID;
 
-        switch (vmBase.getCpuPinningPolicy()) {
+        switch (vmFromParams.getCpuPinningPolicy()) {
         case MANUAL:
-            if (StringUtils.isBlank(vmBase.getCpuPinning())) {
+            if (StringUtils.isBlank(vmFromParams.getCpuPinning())) {
                 result = new ValidationResult(EngineMessage.ACTION_TYPE_CANNOT_SET_MANUAL_PINNING);
             }
             break;
         case RESIZE_AND_PIN_NUMA:
-            if (numaSet) {
+            if (vmFromDb != null
+                    && (vmFromDb.getNumOfCpus() != vmFromParams.getNumOfCpus()
+                            || !vmFromDb.getvNumaNodeList().equals(vmFromParams.getvNumaNodeList()))) {
                 result = new ValidationResult(EngineMessage.ACTION_TYPE_CANNOT_RESIZE_AND_PIN_AND_NUMA_SET);
                 break;
             }
-            boolean singleCoreHostFound = vmBase.getDedicatedVmForVdsList()
```

Comment:
I wonder if we need it now. The scheduler will prevent it.
Yes, it's nicer having it before. But now we do not require pinning to a host.
Also, it is a bit not related to the current PR.

Reply:
That makes sense but the change belongs more to your PR https://github.com/oVirt/ovirt-engine/pull/402